tests/test_doc_examples.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2019 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sys
+import yaml
+from parameterized import parameterized
+import itertools
+
+from c7n.resources import load_resources
+from c7n.exceptions import PolicyValidationError
+from .common import BaseTest
+
+
+def get_doc_examples(doc_folders):
+    policies = []
+    load_resources()
+    for mod in sys.modules.keys():
+        if any(doc_base in mod for doc_base in doc_folders):
+            module = sys.modules[mod]
+            for sub_item in dir(module):
+                cls = getattr(sys.modules[mod], sub_item, None)
+                if isinstance(cls, type):
```

Comment:
this all feels a bit too cute/clever, we already have the provider registry, just walk that instead of poking at everything in sys.modules

Reply:
ah yeah good call. much easier

Reply:
the only downside now is that some docs are parsed multiple times since the child classes that don't override the docstrings will all reference the same parent docs.

Reply:
just remembering a few times when sys module poking resulted in segfaults... some things poke back ;-) I don't think the dupes matter per se in this context, addressing that more directly is just a visited cache with attr/doc origin lookup per class resolution order, but not needed afaics.